### DISCUSSION TITLE
How to overcome 16,000 files limit

### DISCUSSION THREAD

**Discussion by miguel9554** [vercel/vercel#5037] [Help] [answered]
I'm working on a site with Next.js, composed mainly of static pages, but they greatly exceed the [16,000 limit on output files](https://vercel.com/docs/platform/limits#files). Is there a way to overcome this? Is my only option to reduce the number of pages? Deploying Next.js to Vercel is really great, it would be sad to migrate to another plattform because of number of files.

**Comment by styfle** [comment]
Hi @miguel9554 

The limit is not configurable at this time.

However, we are currently rewriting the build pipeline to increase the limit.

**Comment by coetry** [comment, marked as answer]
Hi @miguel9554 . Pre-rendering 16k static pages at build time can take quite a bit of time. It gets worse as the number of pages increase. Have you considered building on a set of your most critical pages and lazily building the rest with `fallback: true` ?

**Reply by miguel9554** [reply]
Yes, finally I discovered that option, it was exactly what I was looking for!

**Comment by maddulakrishna** [comment]
Hi,

I'm also facing the same issue. 

We are using the AZURE CI/CD pipelines to deploy the code into VERCEL. Here is the command, we are using it.



vercel build --prod --token=$(VERCEL_TOKEN)



In our solution , we have some api's and when we run the above command and see the "output" folder, noticed that it's creating the "node_modules" for each api and some other files in that api folder. Attached the screenshot of build output.
![artifact](https://github.com/vercel/vercel/assets/117444169/9fca02cc-a7b5-432f-bbe8-d18267d44b45)





Is it expected? Could you please let me know how to avoid these files in build. Due to this , it's generating 15k+ files and vercel deploy command not allowing these files.



Is it best-practice to use the Azure Devops? (Or) It's good to use the Braching strategy (like deploy the code when push the code to branch)



Please let me know the views.



Thanks,

Krishna
